acoular/calib.py

```diff
@@ -57,3 +118,29 @@ def import_data(self):
             data.append(float(element.getAttribute('factor')))
         self.data = array(data, 'd')
         self.num_mics = self.data.shape[0]
+
+    def result(self, num):
+        """Python generator that processes the source data and yields the time-signal block-wise.
+
+        This method needs to be implemented by the derived classes.
+
+        Parameters
+        ----------
+        num : int
+            This parameter defines the size of the blocks to be yielded
+            (i.e. the number of samples per block)
+
+        Yields
+        ------
+        numpy.ndarray
+            Two-dimensional output data block of shape (num, sourcechannels)
+            where sourcechannels is numchannels if the source data is in the time domain
+            or sourcechannels is numchannels*numfreqs of numchannels*numchannels*numfreqs
+            if the source data is in the frequency domain.
+        """
+        for block in self.source.result(num):
+            if self.data[self.channels].shape[0] == block.shape[1]:
```

Comment:
Can we check this before the generator starts? we could simply check if numchannels fits num_mics - len(invalid_channels)

Reply:
The problem is that this would only work if the source is a time domain source. To account for frequency domain sources, we would need more logic.